src/TensorOneD.hpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef TENSORONED_HPP_INCLUDE
+#define TENSORONED_HPP_INCLUDE
+
+#include "geopm/json11.hpp"
+#include <cstddef>
+
+namespace geopm
+{
+    /// @brief Class to store and perform operations on 1D Tensors,
+    ///        aka vectors, suitable for use in feed-forward neural
+    ///        networks.
+    class TensorOneD
+    {
+        public:
+            ///@brief Constructor
+            TensorOneD();
```

Comment:
```suggestion
            TensorOneD() = default;
```
Then remove the empty implementation in the cpp.

OTOH, If you never intend on using this constructor, delete it with ```TensorOneD() = delete;``` instead.

Reply:
Done